lib/libzfs/libzfs_pool.c

```diff
@@ -3271,11 +3271,29 @@ zpool_vdev_remove(zpool_handle_t *zhp, const char *path)
 		break;
 
 	case EBUSY:
-		zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
-		    "Pool busy; removal may already be in progress"));
+		if (islog) {
+			zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
+			    "Intent log replay needed;\n"
+			    "Mount datasets to replay log."));
+		} else {
+			zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
+			    "Pool busy; removal may already be in progress"));
+		}
 		(void) zfs_error(hdl, EZFS_BUSY, msg);
 		break;
 
+	case EACCES:
+		if (islog) {
+			zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
+			    "Intent log replay needed.\n"
+			    "Mount datasets to load crypto keys "
+			    "and replay log."));
```

Comment:
"Mount encrypted datasets to replay logs" would be more correct here

Reply:
It flows a better to when reading it. @PaulZ-98 can you update the wording have an above.  Then this should be ready.

Reply:
Done!

++ zpool import testpool
++ zpool remove testpool sdc
cannot remove sdc: Mount encrypted datasets to replay logs.